packages/blocks/src/api/factory.js

```diff
@@ -88,8 +88,38 @@ export function createBlocksFromInnerBlocksTemplate(
 }
 
 /**
- * Given a block object, returns a copy of the block object, optionally merging
- * new attributes and/or replacing its inner blocks.
+ * Given a block object, returns a copy of the block object without sanitizing its attributes,
+ * optionally merging new attributes and/or replacing its inner blocks.
+ *
+ * @param {Object} block              Block instance.
+ * @param {Object} mergeAttributes    Block attributes.
+ * @param {?Array} newInnerBlocks     Nested blocks.
+ *
+ * @return {Object} A cloned block.
+ */
+export function duplicateBlock( block, mergeAttributes = {}, newInnerBlocks ) {
```

Comment:
A complete alternative would be a separate pure clone function that accepts the `blockType`, and the widget editor could then declare when the __widgetId attribute should be allowed:
```js
const blockTypeWithWidgetId = withWidgetId( widgetBlockType );
const cloned = cloneBlockUsingBlockType( blockTypeWithWidgetId, widgetBlock );
```